src/_pytest/config/__init__.py

```diff
@@ -86,17 +86,30 @@ class ExitCode(enum.IntEnum):
 
 
 class ConftestImportFailure(Exception):
-    def __init__(self, path, excinfo):
+    def __init__(self, path, excinfo, plugin=None):
         super().__init__(path, excinfo)
         self.path = path
         self.excinfo = excinfo  # type: Tuple[Type[Exception], Exception, TracebackType]
+        self.plugin = plugin  # name of plugin if exception in plugin; None otherwise
 
     def __str__(self):
         return "{}: {} (from {})".format(
             self.excinfo[0].__name__, self.excinfo[1], self.path
         )
 
 
+class PluginImportFailure(ImportError):
+    def __init__(self, plugin, excinfo):
+        super().__init__(plugin, excinfo)
```

Comment:
I think this `super()` call is maybe not right?  I think this puts the wrong data into `ImportError`'s members -- should this extend a different type?

Reply:
I've used ``ImportError`` to avoid problems with these kinds of tests: https://github.com/pytest-dev/pytest/blob/19ad5889353c7f5f2b65cc2acd346b7a9e95dfcd/testing/test_pluginmanager.py#L295
But I think removing that line with ``super()`` won't mess anything. WDYT?

Reply:
It is probably best to pass just a `msg` argument to the parent class, for example:

```python
class PluginImportFailure(ImportError):
    def __init__(self, plugin, excinfo):
        self.plugin = plugin
        self.excinfo = excinfo  # type: Tuple[Type[Exception], Exception, TracebackType]
        super().__init__(str(self))

```